.github/workflows/main.yaml

```diff
@@ -0,0 +1,45 @@
+name: Check and test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        python-version:
+        - "3.8" # minimum required
+        - "3.11" # latest
+        - "3.12-dev" # next
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - run: pip install poetry
+    - uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: poetry
+    - run: poetry install
+    - run: poetry run black --check .
+    - run: poetry run isort --check .
+    - run: poetry run flake8
+    - run: poetry run pytest
```

Comment:
Looking at the CI logs, it seems that poetry is running via the system Python (3.10.6 at this time), and then also installs and launches beets with that python version. Do I read the logs correctly?

If so, probably the order should be changed here to run `setup-python` first, such that `pip install poetry` uses the intended Python?

Reply:
> Looking at the CI logs, it seems that poetry is running via the system Python (3.10.6 at this time), and then also installs and launches beets with that python version. Do I read the logs correctly?

You’re right. This behavior is unexpected but your probably correct that it uses the python version it was installed with. To fix this, I added the line `poetry env use $(which python)`.

> If so, probably the order should be changed here to run `setup-python` first, such that `pip install poetry` uses the intended Python?

Unfortunately, this doesn’t work. Poetry needs to be installed before `setup-python` so we can use `cache: poetry`. (See [this run](https://github.com/geigerzaehler/beets-alternatives/actions/runs/4858676611/jobs/8660367144#step:3:8) for a failure.) We could drop `cache: poetry` and write our own caching instructions but this requires more effort and may easily break.